consensus/leader.go

```diff
@@ -121,14 +134,22 @@ func (consensus *Consensus) onPrepare(recvMsg *FBFTMessage) {
 		}
 	}
 
-	if consensus.decider.IsQuorumAchieved(quorum.Prepare) {
+	signerCount := consensus.decider.SignersCount(quorum.Prepare)
+
+	// check if it is first received signatures
+	// it may multi bls key validators can achieve quorum on first signature
+	hasMultiBlsKeys, isFirstReceivedSignature := consensus.checkFirstReceivedSignature(signerCount, quorum.Prepare)
+
+	quorumPreExisting := consensus.decider.IsQuorumAchieved(quorum.Prepare)
+	//// Read - End
+
+	if quorumPreExisting {
 		// already have enough signatures
 		consensus.getLogger().Debug().
 			Interface("validatorPubKeys", recvMsg.SenderPubkeys).
 			Msg("[OnPrepare] Received Additional Prepare Message")
 		return
```

Comment:
when leader node with multi bls key has quorum, we hit that `return` code and never and will never hit `consensus.didReachPrepareQuorum()`

Reply:
```
if !isFirstReceivedSignature && quorumPreExisting
```
first time with quorum we don't return so we can hit `consensus.didReachPrepareQuorum();` then subsequently we return